site/src/examples/number-input/CustomFormatting.tsx

```diff
@@ -0,0 +1,24 @@
+import { FormField, FormFieldLabel, StackLayout } from "@salt-ds/core";
+import { NumberInput, NumberInputProps } from "@salt-ds/lab";
+
+export const CustomFormatting = () => {
+    const customFormatter: NumberInputProps['format'] = (numStr: string) => {
+      console.log(numStr);
+      if (numStr === '') return '';
+      return new Intl.NumberFormat('en-US', {
+        style: 'currency',
+        currency: 'USD',
+        maximumFractionDigits: 0,
+      }).format(parseFloat(numStr));
```

Comment:
maybe we intend to show this failing since its custom but if you enter '-' on this example it will save as $NaN